<a href="https://colab.research.google.com/github/ksuaray/LAEP_S24/blob/Covid/Covid_Tracker_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Covid-19 Case Tracker**

#**Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that enable us to be able to

In [3]:
!pip install plotly pandas numpy statsmodels ssl scipy

  Using cached ssl-1.16.tar.gz (33 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [22 lines of output]
      Traceback (most recent call last):
        File "/Users/ahmadmardini/Desktop/untitled folder 13/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/ahmadmardini/Desktop/untitled folder 13/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/ahmadmardini/Desktop/untitled folder 13/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_

In [2]:
import pandas as pd
import numpy as np
import plotly.express as px
from IPython.display import Image
import statsmodels.api as sm
import ssl
from scipy import stats


#**Context**

The Associated Press (AP) is using data collected by the Johns Hopkins University Center for Systems Science and Engineering as our source for outbreak caseloads and death counts for the United States and globally.

The Hopkins data is available at the county level in the United States. The AP has paired this data with population figures and county rural/urban designations, and has calculated caseload and death rates per 100,000 people. Be aware that caseloads may reflect the availability of tests -and the ability to turn around test results quickly -- rather than actual disease spread or true infection rates.

This data is from the [Hopkins dashboard](https://www.arcgis.com/apps/dashboards/bda7594740fd40299423467b48e9ecf6) that is updated regularly throughout the day. Like all organizations dealing with data, Hopkins is constantly refining and cleaning up their feed, so there may be brief moments where data does not appear correctly. At [this link](https://github.com/CSSEGISandData/COVID-19/tree/master/csse_covid_19_data), you’ll find the Hopkins daily data reports, and a clean version of their feed.

The AP is updates this dataset hourly at 45 minutes past the hour.

To learn more about AP's data journalism capabilities for publishers, corporations and financial institutions, go [here](http://https/www.ap.org/en-us/formats/data-journalism) or email kromano@ap.org.

Attribution: Johns Hopkins University COVID-19 tracking project

In [7]:
# Replace 'image_url' with the URL of the image you want to display
image_url = 'https://t3.ftcdn.net/jpg/03/33/03/90/240_F_333039083_vkw5CtEmwFSv4ibMFpOpmv49pPTawcDh.jpg'

# Display the image
Image(url=image_url)

#**About the Dataset**

This dataset contains 142 rows corresponding to a random sample of counties. A total of 7 variables are provided as listed below:

**Variables**

| Variable Name(s)  | Description |
|-------------------|-------------|
| County_name       | The name of the county |
| state             | State in which the county is located |
| nchs_urbanization | Urban-Rural category. For more details see [CDC Urban-Rural Classification Scheme for Counties](https://www.cdc.gov/nchs/data/series/sr_02/sr02_166.pdf) |
| total_population  | County population |
| confirmed         | Number of confirmed covid-19 cases in county |
| confirmed_per_100000 | Population adjusted confirmed covid-19 case rate per 100000 people |
| deaths            | Number of deaths in county due to covid-19 |
| deaths_per_100000 | Population adjusted covid-19 death rate per 100000 people |



*Attribution:  FiveThirtyEight.com*

We can view a snippet of the data by first importing it directly from the url below[link text](https://).

**Data**

In [7]:
# This creates an unverified SSL context
ssl._create_default_https_context = ssl._create_unverified_context
file_path = "https://raw.githubusercontent.com/ksuaray/LAEP_S24/Covid/covid_cases23.csv"
df = pd.read_csv(file_path)


Next, we can display the data by typing the name of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [8]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

county_name          state   nchs_urbanization  total_population  \
0       Lowndes        Alabama        Medium metro             10236   
1       Ontario       New York  Large fringe metro            109472   
2      Waukesha      Wisconsin  Large fringe metro            398879   
3      Escambia        Florida        Medium metro            311522   
4    Greenbrier  West Virginia            Non-core             35347   
..          ...            ...                 ...               ...   
137     Bourbon       Kentucky        Medium metro             20144   
138      Schley        Georgia        Micropolitan              5211   
139     Neshoba    Mississippi            Non-core             29376   
140        Wise       Virginia        Micropolitan             39025   
141  Cottonwood      Minnesota            Non-core             11372   

     confirmed  confirmed_per_100000  deaths  deaths_per_100000  
0         3251              31760.45      80             781.56  
1        25821              23586.85     212             193.66  
2       137985              34593.20    1216             304.85  
3        96194              30878.72    1452             466.10  
4        12633              35739.95     182             514.90  
..         ...                   ...     ...                ...  
137       7688              38165.21      73             362.39  
138       1387              26616.77      11             211.09  
139      12475              42466.64     247             840.82  
140      13596              34839.21     225             576.55  
141       3714              32659.16      42             369.33  

[142 rows x 8 columns]

**Assignment 3**

Statistical Inference for One Sample Proportion and Mean

**Instruction**

This assignment is on inferential statistics. Use SPSS to conduct the analysis and complete each of the following questions. As appropriate, copy the SPSS output and paste it into the correct part below. For problems that require a written response, type the answer below.

**QUESTION 1 - LAST NAME: A-L**

Construct and interpret the 98% confidence interval for the true proportion of counties with NCHS Urbanization regions categorized as Large fringe metro. Write up the solution using the PMACC procedure.

Parameter

True Proportion (p): This is the actual proportion of all counties that fall into the "Large fringe metro" category according to the NCHS Urbanization classification. This proportion 

p is what the sample proportion 
$\hat{p}$ is estimating, and the confidence interval is constructed around 
$\hat{p}$ to capture p with a certain level of confidence

Method


Identify the sample size (
n) and the number of successes (
x), where a success is a county being classified as "Large fringe metro".
Calculate the sample proportion 
$\hat{p}$ as $\frac{x}{n}$.
Use the z-score for the desired confidence level (98%) to determine the confidence interval width.

Assumption

Random Sampling: The counties are assumed to be randomly sampled.
Large Sample Size: The sample is large enough for the normal approximation to be valid (typically 
np≥10 and 
n(1−p)≥10).
Independence: Each county's classification is independent of others (valid if sampling without replacement is less than 10% of the population).

Calculation

In [17]:
# Filter for 'Large fringe metro' counties in NCHS Urbanization
large_fringe_metro_counties = df[df['nchs_urbanization'] == 'Large fringe metro']

# Calculate the proportion of counties that are 'Large fringe metro'
total_counties = len(df)
large_fringe_metro_counties_count = len(large_fringe_metro_counties)
p_hat = large_fringe_metro_counties_count / total_counties

# The sample size is the total number of counties
n = total_counties

# Z-score for 98% confidence
Z = stats.norm.ppf(0.99)  # 0.99 because 98% confidence level leaves 1% in the upper tail

# Standard error for the proportion
SE = np.sqrt(p_hat * (1 - p_hat) / n)

# Confidence interval for the proportion
CI_lower = p_hat - Z * SE
CI_upper = p_hat + Z * SE

print(f"98% CI for the proportion of 'Large fringe metro' counties: [{CI_lower:.4f}, {CI_upper:.4f}], p_hat: {p_hat:.4f}")

98% CI for the proportion of 'Large fringe metro' counties: [0.0673, 0.2003], p_hat: 0.1338


Communicate results:

The 98% confidence interval for the proportion of all counties categorized as "Large fringe metro" is calculated as shown. We interpret this interval by stating that based on our sample, we are 98% confident that the actual proportion of counties categorized as "Large fringe metro" in the NCHS Urbanization regions falls between the lower and upper bounds of the interval. This provides a statistically reliable estimate of how widespread the "Large fringe metro" classification is among U.S. counties.

**QUESTION 2**

Is there convincing evidence that the population mean graduate median salary is more than $70,000? Conduct a hypothesis test using α = 0.1. Write up the solution using the PMACC procedure.

Parameter:

Method:

Assumptions:

Calculations:

Communicate results:

Parameter

Population Mean (
μ): This is the parameter we are interested in, specifically the mean graduate median salary across all graduates.

Method

Hypothesis Test:
Null Hypothesis (H0): The population mean graduate median salary is $70,000 or less (μ≤70,000).
Alternative Hypothesis (H 1): The population mean graduate median salary is more than $70,000 (μ>70,000).
We will use a one-sample t-test if the sample size is small or population variance is unknown, and the Z-test if the sample size is large and the population variance is known.

Assumption

Normality: The distribution of the graduate median salary should be approximately normal. This assumption is crucial especially if the sample size is small. If the sample size is large, the Central Limit Theorem allows us to use the t-test regardless of the distribution’s shape.
Independence: The salaries are assumed to be independently collected, with no two salaries influencing each other.
Random Sampling: The data should be collected from a random sample of graduates.


Calculation

In [13]:
# Sample data: number of confirmed cases per 100k
cases_per_100k = df['confirmed_per_100000'].dropna()  # Assume this is the relevant column

# Hypothesized population mean
mu = 30000

# Conduct the one-sample t-test
t_statistic, p_value = stats.ttest_1samp(cases_per_100k, mu)

# Adjust the p-value for a one-tailed test (if testing for greater than)
if t_statistic > 0:
    p_value /= 2
else:
    p_value = 1 - (p_value / 2)

# Output the results
print(f"T-statistic: {t_statistic}")
print(f"P-value: {p_value}")

T-statistic: -0.7477124071896145
P-value: 0.7720605693872986


Communicate results:

The t-statistic of -0.7477124071896145 indicates that the sample mean is lower than the hypothesized population mean under 
H
0​	
  (it is negative because the sample mean is less than $70,000). Since we are conducting a right-tailed test, a negative t-statistic would not lead to rejecting 
H
0
​	
 .

##**QUESTION 3**

Generate a paragraph of at least 100 words to address one of the following questions:  

1. Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major. 

2. Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career.


**QUESTION 1 - LAST NAME: M-Z**

Construct and interpret the 95% confidence interval for the population mean number of deaths in each county. Write up the solution using the PMACC procedure.

Parameter

Population Mean (
μ): The average number of deaths in each county.

Method

Confidence Interval Calculation:
Using a t-distribution to construct the confidence interval for the mean. This is appropriate when dealing with population means and the population standard deviation is unknown.


Assumption

Normality: The number of deaths per county is assumed to be normally distributed, especially important if the sample size is small. If the sample size is large, the Central Limit Theorem supports the normality assumption.
Random Sampling: The data on deaths per county is assumed to be collected randomly.
Independence: The number of deaths in one county is assumed to be independent of the number of deaths in another.

Calculation

In [16]:
num_deaths = df['deaths'].dropna()  # Remove NaN values to ensure accuracy

# Calculate sample mean and standard deviation
sample_mean = np.mean(num_deaths)
sample_std = np.std(num_deaths, ddof=1)  # Use ddof=1 for sample standard deviation
n = len(num_deaths)

# Z-score for 95% confidence
Z = stats.norm.ppf(0.975)  # 0.975 because 95% CI leaves 2.5% in each tail

# Calculate the margin of error
margin_of_error = Z * (sample_std / np.sqrt(n))

# Calculate the confidence interval
CI_lower = sample_mean - margin_of_error
CI_upper = sample_mean + margin_of_error

print(f"95% CI for the mean number of deaths: [{CI_lower:.2f}, {CI_upper:.2f}]")

95% CI for the mean number of deaths: [241.79, 449.39]


Communicate results:

The 95% confidence interval suggests that we are 95% confident that the true average number of deaths in each county lies between approximately 241.79 and 449.39. This range provides an estimate of the mean number of deaths across all counties under consideration.

**QUESTION 2**

Is there convincing evidence that the true proportion of counties with NCHS Urbanization regions categorized as Small metro is different from 20%? Use α=0.05. Write up the solution using the PMACC procedure.

Parameter

Population Proportion (p): This is the proportion of all counties categorized as "Small metro" within the NCHS Urbanization classification.

Method

Null Hypothesis (H0): The true proportion of counties that are categorized as "Small metro" is 20% (
p=0.20).Alternative Hypothesis (H1): The true proportion of counties that are categorized as "Small metro" is not 20% (
p
≠
0.20).
We will use a two-tailed Z-test for the proportion because we want to check if the actual proportion is either less than or greater than 20%.

Assumption

Binomial Distribution: The variable of interest (being a "Small metro" county or not) follows a binomial distribution.
Normal Approximation: The sample size is large enough for the normal approximation to the binomial distribution to be valid. This is generally assumed to be adequate if 
np≥5 and
n(1−p)≥5.
Random Sampling: The sample of counties is randomly selected.
Independence: The categorization of one county as "Small metro" is independent of another.

Calculation

In [18]:
# Filter for 'Small metro' counties in NCHS Urbanization
small_metro_counties_count = df[df['nchs_urbanization'] == 'Small metro'].shape[0]
total_counties = df.shape[0]

# Calculate the sample proportion (p_hat) for 'Small metro'
p_hat = small_metro_counties_count / total_counties
p_0 = 0.20  # Hypothesized population proportion of 'Small metro'

# Sample size
n = total_counties

# Calculating the z-statistic
z = (p_hat - p_0) / ((p_0 * (1 - p_0) / n) ** 0.5)

# Calculating the p-value for a two-tailed test
p_value = 2 * (1 - stats.norm.cdf(abs(z)))

# Print the results
print(f"Z-statistic: {z:.4f}")
print(f"P-value: {p_value:.4f}")

Z-statistic: -2.3917
P-value: 0.0168


Communicate results:

The Z-statistic and P-value are used to determine whether to reject the null hypothesis 
H0, which states that the true proportion of counties categorized as "Small metro" is 20% (
p=0.20).

##**QUESTION 3**

Generate a paragraph of at least 100 words to address one of the following questions:  

1. Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major. 

2. Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career.
